### PR TITLE
Allow to store multiple annotations in cvat

### PR DESCRIPTION
**Pull request by alexanderGerbik**
https://p.daedalean.ai/T6093

For the sake of readability all the work on the ticket is split into two pull requests.

This PR is focused on cvat modification that allows to store multiple annotations for the same task and on preserving the correct behavior of existing features:
- changing database tables structure;
- add single/triple annotation task creation parameter;
- upload/dump/validate selected annotation version of the task;
- update auto-assign algorithm to handle multi-annotation case;
- show annotation version in jobs list and in the inventory file.

### CODE REVIEW

File: cvat-core/src/server-proxy.js [hunk 1]
```diff
@@ -26,6 +26,15 @@
         return new ServerError(message, 0);
     }
 
+    function applyJobSelection(jobSelection, queryParams) {
+        if (!jobSelection) return;
```
Comment (gyakushev) [inline cvat-core/src/server-proxy.js]: nit: don't put `return` on the same line as if

File: cvat-core/src/session.js [hunk 2]
```diff
@@ -565,6 +565,7 @@
                 start_frame: undefined,
                 stop_frame: undefined,
                 sequence_name: undefined,
+                version: undefined,
```
Comment (gyakushev) [inline cvat-core/src/session.js:568]: what does this version refer to? software version, index of annotator, revision... ?
`version` looks like a too generic/overloaded word to use here, not sure what a better word to use, but at least think about it

File: cvat-ui/src/components/actions-menu/dump-submenu.tsx [hunk 3]
```diff
@@ -29,12 +32,13 @@ export default function DumpSubmenu(props: Props): JSX.Element {
...
         dumpers,
         dumpActivities,
+        versionsAmount,
     } = props;
 
     return (
         <Menu.SubMenu key={menuKey} title='Dump annotations'>
-            {
-                dumpers.map((dumper: string): JSX.Element => {
+            {versionSubmenu(versionsAmount,
```
Comment (gyakushev) [inline cvat-ui/src/components/actions-menu/dump-submenu.tsx]: nit: newline after `{`
if you have to split line for parameters, at least put that statement on its own line

File: cvat-ui/src/components/actions-menu/load-submenu.tsx [hunk 4]
```diff
@@ -28,12 +31,13 @@ export default function LoadSubmenu(props: Props): JSX.Element {
...
         onFileUpload,
         disabled,
+        versionsAmount,
     } = props;
 
     return (
         <Menu.SubMenu key={menuKey} title='Upload annotations' disabled={disabled}>
-            {
-                loaders.map((_loader: string): JSX.Element => {
+                {versionSubmenu(versionsAmount,
```
Comment (gyakushev) [inline cvat-ui/src/components/actions-menu/load-submenu.tsx]: ditto

File: cvat-ui/src/components/actions-menu/version-submenu.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { Menu } from 'antd';
+
+export default function versionSubmenu(versionsAmount: number, render: () => JSX.Element): JSX.Element[] {
+    if (versionsAmount === 1) return [render()];
```
Comment (gyakushev) [inline cvat-ui/src/components/actions-menu/version-submenu.tsx]: ditto

File: cvat-ui/src/components/task-page/job-list.tsx [hunk 6]
```diff
@@ -77,7 +83,8 @@ function JobListComponent(props: Props & RouteComponentProps): JSX.Element {
         dataIndex: 'sequenceName',
         key: 'sequenceName',
         className: 'cvat-text-color',
-    }, {
+    }, ...versionColumns,
```
Comment (gyakushev) [inline cvat-ui/src/components/task-page/job-list.tsx]: nit: line break after first comma

File: cvat-ui/src/components/task-page/styles.scss [hunk 7]
```diff
@@ -70,6 +70,10 @@
     }
 }
 
+.validation-report-header div:nth-child(2) {
+  margin-left: auto;
```
Comment (gyakushev) [inline cvat-ui/src/components/task-page/styles.scss]: nit: indent

File: cvat-ui/src/components/task-page/validation-report.tsx [hunk 8]
```diff
@@ -52,7 +53,10 @@ function ValidationReportComponent(props: Props): JSX.Element {
         </Button>
     ;
 
-    let content = report ?
+    let content = loading ?
```
Comment (gyakushev) [inline cvat-ui/src/components/task-page/validation-report.tsx]: the statement gets too complex, split please

File: cvat/apps/engine/ddln/sequences.py [hunk 9]
```diff
@@ -30,11 +30,23 @@ def group(sequences, chunk_size):
     return result
 
 
-def distribute(chunks, assignees):
+def distribute(chunks, assignees, times_assigned=1):
     """Distribute chunks between assignees
 
-    Each assignee gets no more than 1 chunk,
+    If times assigned is equal to 1, each assignee gets no more than 1 chunk,
```
Comment (gyakushev) [inline cvat/apps/engine/ddln/sequences.py]: this function encapsulates two different logics for case of `times_assigned=1` or `>1`
better make two explicit functions

**Review by gyakushev — APPROVED** [review]
Generally great! Some style nits.
I am not very happy with using word `version` in this context, because it is extremely confusing - we have lots of different meanings for this word within the project.

File: cvat-core/src/session.js (hunk 2 above)
Comment (alexanderGerbik) [reply cvat-core/src/session.js:568]: It refers to the index of annotator (but it's a bit more complicated, as one version gets annotated by multiple annotators if the task has thousands of frames).

I agree that version is too generic and I don't like using this word either. I've considered using other words for this term (variants, alternatives, sources, ~options~). But Sveta and David refer to it as _versions_, so I think there would be a lot of confusion if I used another term in the code.

Moreover, there is a ticket for VCS-like versions (https://p.daedalean.ai/T5144). I think that using term `revision` would be the right choice there.

**Pull request merged**